### ISSUE TITLE
Legend is not displayed when name repeats

### ISSUE THREAD

**Issue by Mykhaylo12** [apache/echarts#18242]
### Version

5.4.0

### Link to Minimal Reproduction

https://codepen.io/Mykhailo-codepen/pen/rNrPbeV

### Steps to Reproduce

1)create pie chart with legend
2)add data to legend with elements which has same name.

### Current Behavior

If name of legend repeats it will be omitted and only first will be shown

### Expected Behavior

All names of legend are displeyed even they repeats

### Environment

```markdown
Problem in all browsers independent of Framework
```

### Any additional comments?

![image](https://user-images.githubusercontent.com/55308826/216775265-d4ec3fa9-36d4-4e6a-8038-99ce4c63cae6.png)

**Comment by helgasoft** [comment]
how about a [little cheat](https://echarts.apache.org/examples/en/editor.html?c=map-bar-morph&code=MYewdgzgLgBAJgQyggvAbQFA2z7BvGANwQBsBXAUwC4YBGABgBYAOAGhjAQFtqYByWgDoArAGs-MAL6ssuHAWLleDRgHZ2nHjQEjxUmXPlFSlGsOb0N3Xn3ojme6bMMKTvc7Stb-d86P5S2AD0QTAQAA4IwBQwCHBwFHDOOAC6ANwYIOFQAJbgMCgweLJQICAkueE0BFAATjkA5g0Utdo5UBRcEk7YJBTNYHDV8EgINIjI-rIQLTkUEDSYLsm4mjYAgsDREBAwAGK1IF0GhjBQAJ7hNuFzfCeGtXE5ZAswaHyM9ACkd_yq33wUvc5AhCCAcnAADIIABGFBIAHlCC0SAgqjAAGakGbA3CouEkaorOQQAAWIAA7jQsSQccTcOEQBB2nkwNpomAOrU-PSpqcYJ1wqSEMzXsV-b1YfCiRKcGTKTQ6pRcacMeAoABlHIAL14nxVhjVnIA6hRGqSoNoYeU4DzZZJeT1Tvj4ZCcmBeOKJfKqZjsRRHQaRshxqNiQ7cCkMJI0kA) ?
![image](https://user-images.githubusercontent.com/13038071/216802009-64559ca1-401c-4d67-9fa4-aa99be54f75c.png)

**Comment by Mykhaylo12** [comment]
Thank you! This workaround I will use like last option because:
1) it add spaces to legend and it will be not symmetrical
2) amount of names are dynamic (can be more than 10) and I need to create additional method which handle all this workaround

**Comment by helgasoft** [comment]
...or rethink the data structure. Legend is used for series and categories,
not for values.

**Comment by helgasoft** [comment]
how about [this way](https://echarts.apache.org/examples/en/editor.html?c=map-bar-morph&code=MYewdgzgLgBAJgQyggvAbQFA2z7BvGANwQBsBXAUwC4YBGABgBYAOAGhjAQFtqYByWgDoArAGs-MAL6ssuHAWLleDRgHZ2nHjQEjxUmXPlFSlGsOb0N3Xn3ojme6bMMKTvc7Stb-d845jOOAC6ANwYIAAOUACW4DAoMHiyUCAgJDERNARQAE7RAOb5FDna0VAUXBJO2CQURWBwWfoB2IjIEBRQNGhJuHDRPJCxkN18ipR87HyaFHxBBjgQIGQ5wLxtCLKS87IdeRQQ3YH4xzgz2gCCwGsQEDAAYjkglQuGUACeETYR0bOvcjkEP0yIcYGg-Ix6ABSSb8VTQub_XAAemRCEIIGicAAMggAEYUEgAeUIxRICEyMAAZqQOkicOSCSQmhAABYgADuNFylHYERAEDKw1KQzgf2pIByXCQ5RK_DwAAE0PQgpI-NVDDAyhUAMofWpZPGSsU5ADqWKgrJoACZ2EaciaAMJpSXaEgFVlQfKA97q-nYCoRVkIQWg3qamr4wlZU6GNmc7k5XmxuRU8BQHXRABevEh_tT6dNFA9XX4RpIcD4KewkhTGsMqMZhOx0TAvAIMHjXOptIoNfzqNAJElACF3toNpMU6iNjQNqda7gghhJCEgA), with a dataset 

![image](https://user-images.githubusercontent.com/13038071/217059576-dd44674b-3f1c-4686-bd36-2e9dcfa9918c.png)

**Comment by Mykhaylo12** [comment]
I have requirements. I don't decide by myself

**Comment by ptulou** [comment]
I have a similar case. In my case I've got many series on a chart. Some of the series are named line plots. Some of them show the results of analytics on the data using mark areas, and such. We want to be able to toggle the analytics on and off separately from the data through the legend. But, if the user happens to name the data trend the same thing as the analyitic, we get this behavior where the two are treated as one thing. It would be great if we could supply an id value to the legend data so that the id could be used for uniqueness and the name could just be for display. Ideally, that id value could be the id of the series so that we're linking the two on something more concrete than the user modifyiable name?

**Comment by abbasgalen** [comment]
Agree, this is very helpful to have ability to repeat names in the series. having an id in the legend that links upto the id of the series would be great.

**Comment by PIMBA** [comment]
any news?

**Comment by helgasoft** [comment]
>  keep the 2 legend items with the same name but different color

@vandelpavel - something [like that](https://echarts.apache.org/examples/en/editor.html?c=bar-simple&code=PYBwLglsB2AEC8sDeAoWsAeBBDEDOAXMmurACYCGYFRA2gOQCyM9ANLPQCoCuApmxwDqvMgK4ALbvQC6JAL6sSATxz4iSWApJ5eAJwi9CsWiXSpSpStToBGAEwAGdo6ewbAVlcAOB9MUX0MCUQXiJ6ACMKXQFoCgBbUPoEijxuXV4AOXjeAGoyXgA3CABjTOyOU01_UnMAqxpjT3YbB1cAdm9faosgkLDI6PZYhIIk3hS0soS8wpKp_kq5Ej8SABteAHNeaDICJCW5AG4gA) ? Add a space char.

![image](https://github.com/apache/echarts/assets/13038071/9c31e433-3214-431f-bbcb-e6a1eb6672a1)

**Comment by vandelpavel** [comment]
Nice workaround but I'm developing an app which must render dynamic charts with dynamic data.
I don't know how many series/yaxis/xaxis there will be at the end.

I'm currently patching this issue using a suffix ex: `01/01-01/31`

**Comment by helgasoft** [comment]
>  it eats a lot of horizontal space on labels

using legend.type:['scroll'](https://echarts.apache.org/en/option.html#legend.type) would help

**Comment by vandelpavel** [comment]
My previous screenshot only had 2 series.
I'm already using the type "scroll" for legend. This helps but it still is just a workaround.

![image](https://github.com/apache/echarts/assets/61196133/bbb66fdc-17c6-496a-8604-3032d2a88d26)

Another issue with my specific workaround:
If you need to add a suffix for time intervals then consider to add the year because if you select the same month of 2 different years then the system will break.

**Comment by minhnhat44** [comment]
> Thank you! This workaround I will use like last option because:
> 
> 1. it add spaces to legend and it will be not symmetrical
> 2. amount of names are dynamic (can be more than 10) and I need to create additional method which handle all this workaround

@Mykhaylo12 : Have a better workaround using legend's formatter. Adding suffix for `name`, then using formatter to cut the suffix by delimiter
```option = {
  legend: {
    formatter: function (name) {
        return name.split('-')[0];
    }
  },
  tooltip: {},
  dataset: {},
  xAxis: { type: 'category' },
  yAxis: {
    name: 'Test label',
    nameLocation: 'middle',
    nameGap: 100
  },
  series: [{ type: 'bar',
    name: "Apple-id1",
    data: [1,2,3,4,5,6,7,8,9,10,11,12]
  },
  { type: 'bar',
    name: "Apple-id2",
    data: [1,2,3,4,5,6,7,8,9,10,11,12]
  },
  ]
};

**Comment by vandelpavel** [comment]
@minhnhat44 Be careful with your suggestion.
If the user/db uses a "dashed-name" it will break your logic, ex:
```ts
const name='dashed-name-id1';
const id=name.split('-'); // => "dashed"
```
If you want to use your workaround rather use something like this:
```ts
const name='dashed-name-id1';
const id=name.split('-').slice(-1); // => "id1"
const nameWithoutId=name.split('-').slice(0,-1).join('-'); // => "dashed-name"
```

**Comment by selunders** [comment]
I ran into the same issue, ended up appending zero-width spaces (\u200B)

```js
const mySeries = myData.map((data, index) => ({
  name: `${data.name}${'\u200B'.repeat(index)}`,
  ... // rest of series info
})
```
If you wanted to be extra careful to prevent duplicates:
```js
  name: `${data.name.replace(/\u200B/g, "")}${`\u200B`.repeat(index)}
```
Modifying @helgasoft 's [example from earlier](https://github.com/apache/echarts/issues/18242#issuecomment-1967090865):
![image](https://github.com/user-attachments/assets/f7f1a8c8-e286-4cff-b16c-3db06239ec4c)

**Comment by kianj2** [comment]
I agree with @ptulou. In our use case series names are unpredictable, easily repeated, and customizable by the user. It would be nice if the legend could work with the series ID. Currently, the legend breaks when there are duplicate names.
